WoWResourceParser/Program.cs

```diff
@@ -233,85 +244,108 @@ static void ParseAllADTs(ref HashSet<string> wmos, ref HashSet<string> m2s, ref
             }
         }
 
-        static void ParseAllWMOs(HashSet<string> wmos, ref HashSet<string> m2s, ref HashSet<string> blps, string dataFolder)
+        static void ParseAllWMOs(HashSet<string> wmos, ref HashSet<string> m2s, ref HashSet<string> blps, string dataFolder, string objectFolder)
         {
             var subWmos = new HashSet<string>();
             foreach (var wmoPath in wmos)
             {
                 string fullWmoPath = $"{dataFolder}\\{wmoPath}";
-                if (File.Exists(fullWmoPath))
-                {
-                    ExtractAssetsFromWMO(ref m2s, ref blps, fullWmoPath, dataFolder);
-                    var pathPartA = wmoPath.Substring(wmoPath.LastIndexOf('\\') + 1);
-                    var file = pathPartA.Substring(0, pathPartA.LastIndexOf('.')).ToUpper();
-                    var folder = fullWmoPath.Substring(0, fullWmoPath.LastIndexOf('\\'));
-                    var parentShortFolder = folder.Substring(dataFolder.Length + 1);
-                    if (!Directory.Exists(folder))
-                    {
-                        Console.WriteLine(" [ERROR]: Unable to find path: " + folder);
-                        continue;
-                    }
-                    foreach (var filePath in Directory.EnumerateFiles(folder))
-                    {
-                        var subFilePath = filePath.Substring(filePath.LastIndexOf('\\') + 1);
-                        if (subFilePath.ToUpper().Contains(file))
-                        {
-                            subWmos.Add(parentShortFolder + "\\" + subFilePath);
-                            // Check for x_000.wmo - a sub wmo with no useful data for us
-                            if (!Regex.IsMatch(subFilePath, @"_\d{3}\.\wmo"))
-                            {
-                                ExtractAssetsFromWMO(ref m2s, ref blps, filePath, dataFolder);
-                            }
-                        }
-                    }
-                }
-                else
+                ProcessWMO(dataFolder, wmoPath, fullWmoPath, ref m2s, ref blps, ref subWmos);
+            }
+            if (objectFolder.Length > 0)
+            {
+                foreach (var filePath in Directory.EnumerateFiles(objectFolder, "*.*", SearchOption.AllDirectories))
                 {
-                    Log(" [ERROR] Unable to find file path: " + fullWmoPath);
+                    var wmoPath = filePath.Replace(objectFolder, "");
+                    if (filePath.ToLower().EndsWith("wmo"))
+                        ProcessWMO(dataFolder, wmoPath, filePath, ref m2s, ref blps, ref subWmos);
+
```

Comment:
This works but it feels like this function is doing two different things based on the arguments supplied:
1) Parse all WMOs in the Set parameter
2) Parse all WMOs in the object folder provided

I think it would be cleaner to separate out the logic into different methods, but this entire tool is pretty hacked together so I'm not too worried.